.gitignore

```diff
@@ -4,3 +4,8 @@ target
 # Vim
 *.swp
 Session.vim
+
+# Eclipse
+.classpath
+.project
+.settings/
```

Comment:
I use Eclipse, so I needed this.

Reply:
You can also save personal preferences in `~/.config/git/ignore`

Reply:
Would you like me to remove `.gitignore` from this Pull Request?

Reply:
Oh, I added the vim ignore in there... I did not know about `~/.config/git/ignore` back then. You could remove the vim ones too. It makes sense to keep the Maven one since maven is specific to this project. It's not critical though.